konnectivity-client/pkg/client/client.go

```diff
@@ -123,8 +130,10 @@ func (t *grpcTunnel) serve(c clientConn) {
 					connid: resp.ConnectID,
 				}
 				select {
-				case ch <- result:
-				default:
+				// try to send to the result channel
+				case pendingDial.resultCh <- result:
+				// unblock if the cancel channel is closed
+				case <-pendingDial.cancelCh:
```

Comment:
Nit, the log in this case says ""DialResp has no receiver;" can we change that to something like "Pending dial has been canceled;" which seems more accurate? Having no receiver seems like a necessary but not sufficient condition now.

Reply:
that was based on the documented contract of `cancelCh`:

> cancelCh is the channel closed when resultCh no longer has a receiver

happy to reword to whatever makes sense